File: c/year2015/day01.c
int solve_2015_01_1(char * input)
{
	int floor = 0;
	while (*input) {
		if (*input == '(') floor++;
		if (*input == ')') floor--;
		input++;
	}
	return floor;
}

int solve_2015_01_2(char * input)
{
	int floor = 0;
	int position = 1;
	while (*input) {
		if (*input == '(') floor++;
		if (*input == ')') floor--;
		if (floor < 0) return position;
		position++;
		input++;
	}
	return -1; // unreachable for valid input
}

File: java-aoc/year2015/Day01.java
package year2015;

public class Day01 {
	public static int solvePart1(String input) {
		int floor = 0;
		for (char c : input.toCharArray()) {
			if (c == '(') floor++;
			if (c == ')') floor--;
		}
		return floor;
	}

	public static int solvePart2(String input) {
		int floor = 0;
		int position = 1;
		for (char c : input.toCharArray()) {
			if (c == '(') floor++;
			if (c == ')') floor--;
			if (floor < 0)
				return position;
			position++;
		}
		return -1; // unreachable for valid input
	}
}
